src/tools/kdb/get.c

```diff
@@ -0,0 +1,234 @@
+/**
+ * @file
+ *
+ * @brief KDB get subcommand
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include <colors.h>
+#include <command.h>
+#include <get.h>
+#include <kdbassert.h>
+#include <kdbease.h>
+#include <kdberrors.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define COMMAND_NAME "get"
+
+#define GET_OPTION_KEY(options, name) GET_OPT_KEY (options, COMMAND_BASE_KEY (COMMAND_NAME) "/" name)
+#define GET_OPTION(options, name) GET_OPT (options, COMMAND_BASE_KEY (COMMAND_NAME) "/" name)
+
+void addGetSpec (KeySet * spec)
+{
+	ksAppendKey (spec, keyNew (COMMAND_SPEC_KEY (COMMAND_NAME), KEY_META, "description", "Get the value of an individual key.",
+				   KEY_META, "command", COMMAND_NAME, KEY_END));
+	ksAppendKey (spec, keyNew (COMMAND_SPEC_KEY (COMMAND_NAME) "/all", KEY_META, "description", "Consider all of the keys", KEY_META,
+				   "opt", "a", KEY_META, "opt/long", "all", KEY_META, "opt/arg", "none", KEY_END));
+	ksAppendKey (spec, keyNew (COMMAND_SPEC_KEY (COMMAND_NAME) "/name", KEY_META, "description", "The name of the key", KEY_META,
+				   "args", "indexed", KEY_META, "args/index", "0", KEY_END));
+
+	ADD_BASIC_OPTIONS (spec, COMMAND_SPEC_KEY (COMMAND_NAME))
+}
+
+int execGet (KeySet * options, Key * errorKey)
+{
+	int ret = 0;
+	GET_BASIC_OPTIONS
+
+	bool all = false;
+	tmp = GET_OPTION_KEY (options, "all");
+	if (tmp != NULL)
+	{
+		elektraKeyToBoolean (GET_OPTION_KEY (options, "all"), &all);
+		keyDel (tmp);
+	}
+
+	// required args
+	const char * name = getKeyNameFromOptions (GET_OPTION (options, "name"), errorKey, verbose);
+	if (name == NULL)
+	{
+		RETURN (2)
+	}
+
+	Key * toLookUp = keyNew (name, KEY_END);
+
+	elektraFree ((void *) name);
+
+	KeySet * searchIn = ksNew (0, KS_END);
+	KDB * handle = kdbOpen (NULL, errorKey);
+
+	if (toLookUp == NULL)
+	{
+		ELEKTRA_SET_VALIDATION_SEMANTIC_ERRORF (errorKey, "'%s' is not a valid key name", keyName (toLookUp));
+		keyDel (toLookUp);
```

Comment:
What about moving all the free memory code to `cleanup` section and check there if the variable is NULL and if not free the memory?